### DISCUSSION TITLE
fastapi.security.APIKeyHeader auth class doesn't validate authentication scheme_name argument

### DISCUSSION THREAD

**Discussion by navhits** [fastapi/fastapi#8676] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Security, Depends
from fastapi.security.api_key import APIKeyHeader, APIKey


"""
We want the APIKeyHeader class to return the value of the provided header.
Along with that, we will also need the class to validate if the header's auth scheme matches the provided scheme.
"""


scheme="Token"

auth_header = APIKeyHeader(name="Authorization", scheme_name=scheme, auto_error=False)


async def get_auth(auth_header: str = Security(auth_header)):
    print(f'Original scheme is {scheme}')
    print(f'User provided scheme was {auth_header.split(" ")[0]}')
    if auth_header:
        return auth_header
    else:
        return None

app = FastAPI()


@app.get("/")
async def hello(api_key: APIKey = Depends(get_auth)):
    return "Hello"


if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8000)
```

### Description

* Run the curl command,
    ```sh
    curl --location --request GET 'http://localhost:8000/' \
    --header 'Authorization: Token asd'
    ```
* If you check the console you will see, 
    ```
    Original scheme is Token
    User provided scheme was Token
    ```
* Now change the Auth scheme to Basic or something else and run the same command
    ```sh
    curl --location --request GET 'http://localhost:8000/' \
    --header 'Authorization: Basic asd'
    ```
* If you check the console you will see, 
    ```sh
    Original scheme is Token
    User provided scheme was Basic
    ```
* I expect [`fastapi.security.APIKeyHeader`](https://github.com/tiangolo/fastapi/blob/master/fastapi/security/api_key.py#L41) class to validate the provided auth scheme. Since it takes the arg. `scheme_name` I safely assume its best to validate it within the class rather than me writing the validation.
* Since this is already implemented in [`fastapi.security.HTTPBasic`](https://github.com/tiangolo/fastapi/blob/master/fastapi/security/http.py#L53) in [line 71](https://github.com/tiangolo/fastapi/blob/master/fastapi/security/http.py#L71)

### Operating System

macOS

### FastAPI Version

0.70.0

### Python Version

3.9.4

### Additional Context

I see that in [line 53](https://github.com/tiangolo/fastapi/blob/master/fastapi/security/api_key.py#L53) of [`fastapi.security.APIKeyHeader`](https://github.com/tiangolo/fastapi/blob/master/fastapi/security/api_key.py#L41) class, the usage of `self.__class__.__name__` when `scheme_name` is not provided, might cause confusion when implementing the validation of scheme. It is safe to either return the value and skip the scheme validation or simply raise Exception to the user.

**Comment by navhits** [comment]
I also have another suggestion for usage of scheme_name in [APIKeyQuery](https://github.com/tiangolo/fastapi/blob/master/fastapi/security/api_key.py#L19) and in [APIKeyCookie](https://github.com/tiangolo/fastapi/blob/master/fastapi/security/api_key.py#L80). This argument won't be necessary as auth schemes are usually not part of query params or cookies according to my knowledge.

**Comment by navhits** [comment]
I'd be happy to send a PR if necessary. Let me know ;)

**Comment by YuriiMotov** [comment, marked as answer]
`scheme_name` parameter has different meaning, it's just the name that will be used in openapi for this security scheme:

```
    "securitySchemes": {
      "APIKeyCookie": {
        "type": "apiKey",
        "in": "cookie",
        "name": "session"
      }
    }
```

`APIKey***` classes assume that token is sent as a value of parameter without any prefixes.
If you need to accept it as `Authorization: Schema asd` then you need to use another security class
